src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -34,11 +34,19 @@ class AddToCollectionModal extends React.Component {
 	}
 
 	async componentDidMount() {
-		const collections = await this.getCollections();
+		const collections = this.props.entityType ? await this.getCollections() : [];
 		// eslint-disable-next-line react/no-did-mount-set-state
 		this.setState({collectionsAvailable: collections});
 	}
 
+	async componentDidUpdate(prevProps) {
+		if (prevProps.entityType !== this.props.entityType) {
+			const collections = this.props.entityType ? await this.getCollections() : [];
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({collectionsAvailable: collections});
```

Comment:
`collections` could potentially be `undefined` here (admittedly in rare cases where a call to `getCollections` fails).

If so, further down a check on `this.state.collectionsAvailable.length` will throw an error `Cannot read property 'length' of undefined`.

So we either have to ensure `collectionsAvailable` is never set to `undefined`, or check if it is `undefined` wherever we use it.
I suggest this simple method :
```suggestion
			this.setState({collectionsAvailable: collections || []});
```

And the same should apply for `componentDidMount`.